src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/ITinkerStationRecipe.java

```diff
@@ -40,12 +41,11 @@ default ItemStack assemble(ITinkerStationContainer inv) {
 
   /**
    * Gets the recipe result, or an object containing an error message if the recipe matches but cannot be applied.
-   * TODO 1.20: switch return type to {@code RecipeResult<LazyToolStack>}
    * @return Validated result
    */
-  default RecipeResult<ItemStack> getValidatedResult(ITinkerStationContainer inv) {
-    ItemStack result = assemble(inv);
-    if (result.isEmpty()) {
+  default RecipeResult<LazyToolStack> getValidatedResult(ITinkerStationContainer inv) {
```

Comment:
The key idea with the switch to LazyToolStack is we want to also pass it in instead of `ItemStack result` to `updateInputs` as that saves us needing to create the tool stack again just to figure out how to update inputs.